pmda/test/test_util.py

```diff
@@ -130,3 +131,55 @@ def test_make_balanced_slices_ValueError(n_frames, n_blocks,
     with pytest.raises(ValueError):
         make_balanced_slices(n_frames, n_blocks,
                              start=start, stop=stop, step=step)
+
+
+@pytest.mark.parametrize('n_frames', [3, 4, 10, 19, 101, 331, 1000])
+def test_second_order_moments(n_frames):
+    # generate array of random positions in range [-100, 100] for 100 time steps
+    pos = 200*(np.random.random(size=(n_frames, 1000, 3)) - 0.5)
+    # random splitting point
+    isplit = (np.random.randint(1, n_frames-1))
+    # split into two partitions; should be fixed to split by n_blocks partitions
+    p1, p2 = pos[:isplit], pos[isplit:]
+    # create [t, mu, M] lists
+    S1 = [len(p1), p1.mean(axis=0), sumofsquares(p1)]
+    S2 = [len(p2), p2.mean(axis=0), sumofsquares(p2)]
+    # run lists through second_order_moments
+    result = second_order_moments(S1, S2)
+    # compare result to calculations over entire pos array
+    assert result[0] == len(pos)
+    assert_almost_equal(result[1], pos.mean(axis=0))
+    assert_almost_equal(result[2], sumofsquares(pos))
+
+
+@pytest.mark.parametrize('n_blocks', [2, 3, 4, 5, 10, 100, 500])
+@pytest.mark.parametrize('n_frames', [1000, 10000, 50000, 100000])
+def test_reduce(n_frames, n_blocks):
+    # generate array of random positions in range [-100, 100] for 100 time steps
+    pos = 200*(np.random.random(size=(n_frames, 1000, 3)) - 0.5).astype(np.float64)
+    # generate array of indices to split pos by
```

Comment:
Can you use `util.make_balanced_slices()` instead? Or do I misunderstand what you're doing here? You can also ask @VOD555 !

Reply:
I can use util.make_balanced_slices(), but it will create mostly even-length arrays. The current method splits the array into segments of many different lengths, so it just depends on what is preferable. If similar-length arrays are preferable it can easily be adapted over to use util.make_balanced_slices().

Reply:
Stick with your test with random-length slices. Put a comment saying what the intention is.

See comment below for how to streamline the code a bit.